<launch>
    <arg name="initial_pose_a" default="0.0"/>
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="move_forward_only" default="false"/>


    <include file="$(find turtlebot3_slam)/launch/turtlebot3_slam.launch"/>

    <arg name="map_folder" default="map"/>
    <node name="map_saver" pkg="map_server" type="map_saver" args="-f $(find com2009_team61)/launch/maps/map">
        <remap from="/map" to="$(arg map_folder)"/>
    </node>

    <!-- Map server -->
    <arg name="map_file" default="$(find com2009_team61)/launch/maps/map.yaml"/>
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
    </node>

    <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/>

    <node pkg = "com2009_team61" type = "mazeExecutePath.py" name = "maze_node" output = "screen">
    </node>
    <node pkg = "com2009_team61" type = "mapsaver.py" name = "map_saver_seconds" output = "screen">
    </node>

    <arg name="cmd_vel_topic" default="/cmd_vel" />
    <arg name="odom_topic" default="odom" />
    <include file="$(find turtlebot3_navigation)/launch/move_base.launch">
        <arg name="model" value="$(arg model)" />
        <arg name="move_forward_only" value="$(arg move_forward_only)"/>
    </include>
</launch>